tutorials/shading/your_first_shader/what_are_shaders.rst

```diff
@@ -0,0 +1,61 @@
+.. _doc_what_are_shaders:
+
+What are shaders
+================
+
+Introduction
+------------
+
+So, you have decided to give shaders a try. You have likely heard that they can be used to 
+create interesting effects that run incredibly fast. You have also likely heard that they
+are terrifying. Both are true. 
+
+Shaders can be used to create a wide range of effects (in fact everything drawn in a modern
+rendering engine is done with shaders). 
+
+Writing shaders can also be very difficult for people unfamiliar with them. Godot tries to make writing
+shaders a little easier by exposing many useful built-in features and handling some of the
+lower-level initialization work for you. However, GLSL (the OpenGL Shading Language, which Godot uses)
+is still unintuitive and restricting, especially for users who are used to GDScript.
+
+But what are they
+-----------------
+
+Shaders are a special kind of program that runs on the GPU. 
+
+Objects are drawn using an associated shader. The output of the shader is the pixel colors drawn
+to the viewport. Shaders do not save their results, nor can they be used to draw outside their
+corresponding mesh. The lack of flexibility allows shaders to be incredibly fast. 
```

Comment:
"mesh" hasn't been defined yet, and doesn't necessarily apply (2D).

"This lack of flexibility..."